public/app/percona/entitlements/EntitlementsPage.tsx

```diff
@@ -48,20 +51,20 @@ const EntitlementsPage: FC = () => {
   return (
     <PageWrapper pageModel={PAGE_MODEL} dataTestId="page-wrapper-entitlements">
       <PlatformConnectedLoader>
-        {pending ? (
-          <Spinner className={styles.loader} />
-        ) : (
-          data.map((entitlement: Entitlement) => {
-            const { number, name, endDate } = entitlement;
-            return (
-              <div key={number} className={styles.collapseWrapper}>
-                <CollapsableSection label={<Label name={name} endDate={endDate} />} isOpen={false}>
-                  <SectionContent entitlement={entitlement} />
-                </CollapsableSection>
-              </div>
-            );
-          })
-        )}
+        <Overlay dataTestId="entitlements-loading" isPending={pendingRequest}>
+          <PageContent hasData={data.length > 0} emptyMessage={Messages.noData} loading={pendingRequest}>
+            {data.map((entitlement: Entitlement) => {
```

Comment:
`Overlay` already has a `isPending` which hides the children until that's false, but `PageContent` has a `loading` as well. If `PageContent` is only shown if `pendingRequest=false`, will `loading` ever be true?

Reply:
Yes it will be, because Overlay is not hiding children when its loading. It just blur the background.

Reply:
Damn you're right, good one 👍